sns/parity_math.c

```diff
@@ -215,9 +227,34 @@ static void submatrix_construct(struct m0_matrix *in_mat,
 static void dependency_bitmap_prepare(struct m0_sns_ir_block *f_block,
 				      struct m0_sns_ir *ir);
 
+#if ISAL_ENCODE_ENABLED
+/**
+ * Converts m0_bufvec to m0_buf and vice versa.
+ * @param bvec  - Pointer to buffer vector.
+ * @param buf   - Pointer to buffer.
+ * @param count - Number of buffers to copy
+ * @param dir   - Decides direction of conversion.
+ *                True  - Copy data from buffer vector to buffer
+ *                False - Copy data from buffer to buffer vector
+ */
+static void bufvec_buf(struct m0_bufvec *bvec, struct m0_buf *buf,
```

Comment:
I don't like one single function to do buf copy with different direction.

Can you split it to two different functions?
e.g. bufvec2buf() and buf2bufvec()

Reply:
Done. Created two functions bufvec_to_buf_copy() and buf_to_bufvec_copy() to do the operations.